agent/main.go

```diff
@@ -97,12 +98,31 @@ func logError(msg string, err error, cfg *agentConfig) {
 	}
 }
 
-func updateAgents(cfg *agentConfig, cancel <-chan interface{}) {
-	// Self-update
+func agentManifestURL(template string, cfg *agentConfig) string {
 	agentPollURL, err := text.ResolveString(cfg.AgentPollURLTemplate, cfg)
 	if err != nil {
 		log.Fatal("invalid URL template: ", err)
 	}
+
+	// Propagate the cri-endpoint to service.
+	if cfg.CRIEndpoint != "" {
+		url, err := url.Parse(agentPollURL)
+		if err != nil {
+			log.Fatal("couldn't parse agent URL: ", err)
```

Comment:
Should we instead try to parse the `AgentPollURLTemplate`? Would that make sense?

Reply:
It's not obvious the template string is a valid URL and can change in the future, seems safer that way.